### ISSUE TITLE
Consider yats.vim instead of typescript-vim

### ISSUE THREAD

**Issue by sQVe** [sheerun/vim-polyglot#428]
_First and and foremost, thank you for an awesome plugin. It really saves us all A LOT of time :pray:_

I've noted that there is atleast one large syntax issue with `leafgarland/typescript-vim`. It is triggered when writing lambdas / rocket functions with generics like so:

```typescript
const head = <T>(arr: T[]): T => arr[0]
```

Above code snippet breaks highlighting for all code below it. `HerringtonDarkholme/yats.vim` does not have this issue and seems to give better highlighting over the board.

What's the reasoning for opting for `leafgarland/typescript-vim` over `HerringtonDarkholme/yats.vim`?

**Comment by sheerun** [comment]
I tried to use `HerringtonDarkholme/yats.vim` and it seems to have the same issue

**Comment by sheerun** [comment]
The issue seems to be with vim-jsx-pretty, not leafgarland/typescript-vim

**Comment by sheerun** [comment]
I've created issue there: https://github.com/MaxMEllon/vim-jsx-pretty/issues/86

**Comment by gbrlsnchs** [comment]
IIRC, https://github.com/HerringtonDarkholme/yats.vim has bad indentation, although syntax is superior to https://github.com/leafgarland/typescript-vim.

**Comment by sheerun** [comment]
If vim-jsx-pretty fixes this issue in say a month I won't change provider to yats because as mentioned it has worse support for jsx than vim-jsx-pretty.

Maybe suggest yats to integrate with vim-jsx-pretty? Then I'll switch for sure.

**Comment by sQVe** [comment]
@sheerun Nice find! 

The best possible solution would be, like you suggested, to integrate jsx-pretty with yats.

**Comment by sheerun** [comment]
Because there were additional issues with current typescript syntax (#431) I've decided to switch provided to yats, but at the same time leave vim-jsx-pretty so this issue is still present but should probably be fixed in near future.

**Comment by sQVe** [comment]
@sheerun I introduced this issue mainly because of the broken highlighting for rocket functions with generics (`const head = <T>(arr: T[]): T => arr[0]`). This issue was later tracked down to being a problem with `vim-jsx-pretty` (https://github.com/MaxMEllon/vim-jsx-pretty/issues/86). The other issues mentioned here, #431 which became $432, was due to a `styled-components` plugin.

What other reasons was there behind opting for `yats.vim`, except you getting understandingly tired of all these TypeScript issues? Both providers, `yats.vim` and `typescript.vim`, have their upsides and downsides. After using both for quite some time I would probably consider `typescript.vim` to be more stable but less fancy. `yats.vim` wins in highlighting and that it is, at the very least, active in regards of PRs etc.
